tests/unit/test_weights.py

```diff
@@ -207,13 +213,58 @@ def test_simplex_weights(self) -> None:
 
         self.assertTrue(jnp.allclose(output, expected_output, rtol=1e-4))
 
+    def test_calculate_simplex_weights_approximate(self) -> None:
+        """
+        Test the approximate calculation of weights via the simplex method for.
+
+        Since the approximate solution method currently just calls the exact solution
+        method, this test should give exactly the same output (with a warning raised)
+        as the test `test_simplex_weights` above.
+        """
+        # Setup data
+        x = jnp.array([[0, 0], [1, 1], [2, 2]])
+        y = jnp.array([[0, 0], [1, 1]])
+
+        w2 = (
+            -1
+            - 2 * jnp.exp(3)
+            + 3 * jnp.exp(4)
+            + cmath.sqrt(
+                1
+                + 4 * jnp.exp(3)
+                - 6 * jnp.exp(4)
+                + 28 * jnp.exp(6)
+                - 6 * jnp.exp(7)
+                - 21 * jnp.exp(8)
+            )
+        ) / (6 * (jnp.exp(4) - jnp.exp(3)))
+        w2 = jnp.real(w2)
+        w1 = 1 - w2
+        expected_output = jnp.asarray([w1, w2])
+
+        optimiser = coreax.weights.MMD(kernel=coreax.kernel.SquaredExponentialKernel())
+
+        # Solve for the weights
+        with warnings.catch_warnings(record=True) as warning_result:
```

Comment:
Do you want to verify that a warning is raised here, or just capture the warning so it doesn't appear in the pytest log?